src/test/kotlin/com/exactpro/th2/check1/rule/check/TestCheckRuleTask.kt

```diff
@@ -348,6 +349,97 @@ internal class TestCheckRuleTask : AbstractCheckTaskTest() {
+                                            "D" to "4".toValueFilter()
+                                    ))
+                                }.build()
+                            }.build())
+                            addValues(ValueFilter.newBuilder().apply {
+                                messageFilter = messageFilter().apply {
+                                    putAllFields(mapOf(
+                                            "A" to "1".toValueFilter(),
+                                            "B" to "2".toValueFilter()
+                                    ))
+                                }.build()
+                            }.build())
+                        }.build()).build())
+            }.build()
+        }.build()
+
+        val eventID = EventID.newBuilder().setId("root").build()
+
+        checkTask(messageFilterForCheckOrder, eventID, streams).begin()
+
+        val eventBatches = awaitEventBatchRequest(1000L, 2)
+        val eventList = eventBatches.flatMap(EventBatch::getEventsList)
+        assertAll({
+            assertEquals(3, eventList.size)
+        }, {
+            val verificationEvent = eventList.find { it.type == "Verification" }
+            assertNotNull(verificationEvent) { "Missed verification event" }
+
+            val verification = jacksonObjectMapper().readValue<List<Verification>>(verificationEvent.body.toByteArray()).firstOrNull()
+            assertNotNull(verification) { "Verification event does not contain the verification" }
+            val actualLegs = verification.fields["legs"]?.fields?.values?.toList()
+            assertNotNull(actualLegs) { "Actual legs is missed" }
+
+            val expectedLegs = linkedMapOf(
+                    0 to linkedMapOf(
+                            "A" to VerificationStatus.NA,
+                            "B" to VerificationStatus.NA,
+                            "C" to VerificationStatus.FAILED,
+                            "D" to VerificationStatus.FAILED
+                            
+                    ),
+                    1 to linkedMapOf(
+                            "C" to VerificationStatus.NA,
+                            "D" to VerificationStatus.NA,
+                            "A" to VerificationStatus.FAILED,
+                            "B" to VerificationStatus.FAILED
+                    )
+            )
+
+            expectedLegs.forEach { (leg, verificationEntryByField) ->
+                val actualLeg = actualLegs[leg].fields
+                assertNotNull(actualLeg) { "The validation event does not contain the expected leg" }
+                verificationEntryByField.forEach { (field, status) ->
+                    val expectedVerificationEntry = actualLeg[field]
+                    assertNotNull(expectedVerificationEntry) { "Actual leg does not contain the expected field" }
+                    assertEquals(status, expectedVerificationEntry.status)
+                }
+            }
```

Comment:
I have already so code like this in one of the tests. Maybe we should extract the logic into a separate method?